File: app/src/main/java/com/cookplan/recipe/import_recipe/approve_result/ImportRecipeView.java
package com.cookplan.recipe.import_recipe.approve_result;

import com.cookplan.models.Ingredient;
import com.cookplan.models.Recipe;

import java.util.List;
import java.util.Map;

/**
 * Created by DariaEfimova on 08.06.17.
 */

public interface ImportRecipeView {

    void setImportResult(Recipe recipe, Map<String, List<Ingredient>> ingredientList);

    void setError(String s);

    void setRecipeSavedSuccessfully(Recipe recipe);

    void setIngredientSavedSuccessfully(String key);
}

File: app/src/main/java/com/cookplan/recipe/import_recipe/approve_result/ImportRecipeView.kt
package com.cookplan.recipe.import_recipe.approve_result

import com.cookplan.BaseView
import com.cookplan.models.Ingredient
import com.cookplan.models.Recipe

/**
 * Created by DariaEfimova on 08.06.17.
 */

interface ImportRecipeView : BaseView {

    fun setImportResult(recipe: Recipe, recipeToingredientsMap: MutableMap<String, List<Ingredient>>)

    fun setError(s: String)

    fun setRecipeSavedSuccessfully(recipe: Recipe)

    fun setIngredientSavedSuccessfully(key: String)
}
